batch2/batch/worker.py

```diff
@@ -212,12 +212,19 @@ def step(self, name, **kwargs):
 
             async with self.step('runtime', state=None):
                 async with worker.cpu_sem(self.cpu_in_mcpu):
+                    if self.name == 'main':
+                        start_time = time.time()
+                        await self.job.mark_started(worker, start_time)
+
                     async with self.step('starting'):
                         await docker_call_retry(self.container.start)
 
                     async with self.step('running'):
                         await docker_call_retry(self.container.wait)
 
+                    if self.name == 'main':
+                        self.job.mark_ended(time.time())
```

Comment:
This won't run if e.g. wait throws an exception.  We need to capture the time in that case, too.  Also, I think I'd just let the mark complete event set the time instead of sending a separate mark time event.  I think the main runtime timing is actually what we want for the attempt.

Reply:
ok. so you're proposing that we mark the start_time, but the end_time is start_time + main_container_status['runtime']['duration']? That's fine, but I always worry about getting the times to be in the same timezone/units. I'll just have to double check.

Reply:
There are now start_time and end_time (and duration) in the container status timing structs.  Start and end time are in unix time.  So just take the end time.  All times should be in unix time which is seconds elapsed since the epoch, midnight Jan 1 1970 GMT/UTC, so there is no issue with timezones anywhere.